src/client/kinto_http.rs

```diff
@@ -185,25 +198,41 @@ mod tests {
                         }
                     ],
                     "timestamp": 42
-                }"#,
-            );
-        });
-
-        let res =
-            get_latest_change_timestamp(&mock_server_address, "main", "url-classifier-skip-urls")
-                .unwrap();
+                }"#
+                .as_bytes()
+                .to_vec(),
+                response_headers: Headers::new(),
+            }],
+        ));
+
+        let res = get_latest_change_timestamp(
+            &test_client,
+            "https://example.com",
+            "main",
+            "url-classifier-skip-urls",
+        )
+        .unwrap();
 
         assert_eq!(res, 9173);
-
-        get_latest_change_mock.delete();
     }
 
     #[test]
     fn test_bad_url() {
         init();
 
+        let test_client: Box<dyn Requester + 'static> = Box::new(TestHttpClient::new(
+            true,
```

Comment:
This looks like the only place where we passe the *must fail* as true. 

Can't we find a way to avoid it? 
The original test was mainly testing that the error about bad URL was human-readable.